test/integration/targets/apt/tasks/apt.yml

```diff
@@ -112,6 +112,46 @@
     that:
         - "not apt_result.changed"
 
+# Download only
+- name: pre-check absence of downloaded file
+  find:
+    path: "/var/cache/apt/archives/"
+    patterns: 'hello_*'
+  register: pre_apt_download
+
+- name: Clean up hello dpg if it exists
```

Comment:
Looks like for tests to pass, this should be:

```
- name: Clean up hello dpg if it exists
  file:
    path: "{{ item.path }}"
    state: absent
  with_items: "{{ pre_apt_download.files }}"
  when: pre_apt_download.matched >= 1
```

Reply:
Thanks Paul, This pull request has kind of fallen to the wayside, but I'll make some time to test your suggestion :)

Reply:
We've been using your version of the apt module internally with great success, thanks for your contribution. Would love to see it merged into mainline.

Reply:
Thanks for the testing, I'm glad the code is getting some use :) 
fyi, to clarify on the issues with fixing the tests, the biggest issue is the [upgrade.yml](test/integration/targets/apt/tasks/upgrade.yml) tests, specifically around trying to dynamically find a package to test `upgrade:` along with `download_only:`.  
  
Initial tests of the existing upgrade.yml process fails, presumably something to do with how it removes apt sources and references locally to test the upgrade.. so somehow stopping it from "downloading" the already local deb into the apt cache...  
My initial work around tests have been a pain as I really don't want to commit an old deb file to then `apt-get update;upgrade --download-only` to test, nor do I want to use some dodgy 3rd party repo that contains old installers to accommodate getting an old version of an existing application installed to then update.  
  
Last I remember I was looking at making use of the test images being out of date, grabbing one of the pending updates and running it, but this opens up inconsistencies in what actually gets test updated if the base image ever changes.... and along with all of these decisions I've also been very tempted to just re-write all of the apt tests into a more consistent structure as I've already found a handful of invalid tests / logical errors which either result in incorrect confirmations, or incorrect errors... 
  
Bit of a long explanation, but as you've asked, that's the current state of the tests. The downside I guess with adding in functionality that can be used with pretty much every `state:` o.0

Reply:
Hey,
Finally picking this up. Thanks for spotting the error on this test, I've added your suggestion and I'm reviewing the situation with the upgrade tests. Will see if it's accepted without